app/src/main/java/com/pvp/app/Activity.kt

```diff
@@ -38,4 +49,51 @@ class Activity : ComponentActivity() {
             Route.routesUnauthenticated
         }
     }
+
+    private fun isNotificationEnabled(context: Context): Boolean {
+        val enabled = NotificationManagerCompat
+            .from(context)
+            .areNotificationsEnabled()
+
+        if (!enabled) {
+            return false
+        }
+
+        val notificationManager =
+            context.getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
+
+        val channels = notificationManager.notificationChannels
+
+        for (channel in channels) {
+            if (channel.importance == NotificationManager.IMPORTANCE_NONE) {
+                return false
+            }
+        }
+
+        return true
+    }
+
+    private fun openNotificationSettingsForApp(context: Context) {
+        val intent = Intent().apply {
+            action = Settings.ACTION_APPLICATION_DETAILS_SETTINGS
+            addCategory(Intent.CATEGORY_DEFAULT)
+            data = Uri.parse("package:" + context.packageName)
+        }
+
+        context.startActivity(intent)
+    }
+
+    private fun showNotificationPermissionDialog(context: Context) {
+        AlertDialog.Builder(context)
```

Comment:
Can leave as it is, since technically it is a path into inner Builder class's constructor, or can split after AlertDialog. I guess it is better to not split. But writing the idea behind why.